packages/extension/src/background/handlers/Extension.ts

```diff
@@ -195,52 +195,52 @@ export default class Extension {
   }
 
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  public async handle (id: string, type: keyof ResponseTypes, request: RequestTypes[typeof type], port: chrome.runtime.Port): Promise<ResponseTypes[typeof type]> {
+  public handle (id: string, type: keyof ResponseTypes, request: RequestTypes[typeof type], port: chrome.runtime.Port): Promise<ResponseTypes[typeof type]> {
     switch (type) {
       case 'authorize.approve':
-        return this.authorizeApprove(request as RequestAuthorizeApprove);
+        return Promise.resolve(this.authorizeApprove(request as RequestAuthorizeApprove));
```

Comment:
Is this extra resolve needed? This is just a lot of boilerplate that async takes care of for you.

Reply:
I thought the same but eslint was complaining

```
/polkadot-extension/packages/extension/src/background/handlers/Extension.ts                                               
  198:3  error  Async method 'handle' has no 'await' expression  @typescript-eslint/require-await  
```

Reply:
Sorry, only saw that now - yes, the 2.x version now does - in these cases I just silence it since it actually has a valid reason